app/uk/gov/hmrc/platopsgithubproxy/scheduler/GitHubRateLimitMetricsScheduler.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.platopsgithubproxy.scheduler
+
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import akka.actor.ActorSystem
+import cats.implicits._
+import com.kenshoo.play.metrics.Metrics
+import play.api.inject.ApplicationLifecycle
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.lock.{LockService, MongoLockRepository}
+import uk.gov.hmrc.mongo.metrix.{MetricOrchestrator, MetricSource, MongoMetricRepository}
+import uk.gov.hmrc.platopsgithubproxy.config.{GitHubConfig, SchedulerConfigs}
+import uk.gov.hmrc.platopsgithubproxy.connector.GitHubConnector
+import uk.gov.hmrc.platopsgithubproxy.helpers.SchedulerUtils
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+
+
+
+@Singleton
+class GitHubRateLimitMetricsScheduler @Inject()(
+  gitHubConnector: GitHubConnector,
+  schedulerConfig: SchedulerConfigs,
+  gitHubConfig   : GitHubConfig,
+  mongoLock      : MongoLockRepository,
+  metrics        : Metrics,
+  mongoComponent : MongoComponent
+)(implicit
+  actorSystem         : ActorSystem,
+  applicationLifecycle: ApplicationLifecycle
+) extends SchedulerUtils {
+
+  final val metrixLock: LockService = LockService(mongoLock, "metrix-lock" , 20.minutes)
+
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  implicit val ec: ExecutionContext = actorSystem.dispatchers.lookup("scheduler-dispatcher")
```

Comment:
You can just use an injected ExecutionContext here - the scheduler here is unlikely to use up available threads